gradle-diff-task-dependencies/demo/build.gradle

```diff
@@ -0,0 +1,18 @@
+apply plugin: 'java'
+
+task('android-test') << {
+    println 'this is an android test task'
+}
+
+task('apple-test') << {
+    println 'this is an apple test task'
+}
+
+task('all-test') << {
+    println 'this is a test for everything'
+}
+
+task('selectiveTest')
+        .ifDiffMatches(/android\/.*/).dependsOn('android-test')
```

Comment:
What I want to build. This PR is mostly oriented to the API: is this good or should I take a different approach?